### ISSUE TITLE
Ui Panels appearing inside eachother

### ISSUE THREAD

**Issue by EuphieEuphoria** [Yellow-Dog-Man/Resonite-Issues#2141]
### Describe the bug?

When close enough, UI panels will appear to merge with one another.

### To Reproduce

Open an inspector on a slot, Open the component browser and look from an angle or distance.

### Expected behavior

The component browser should appear in front of the inspector

### Screenshots

![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/101027344/d8aada4f-8f18-4524-98b7-a98f322f1650)
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/101027344/88a31568-0655-4352-9889-f551b17b4b50)

### Resonite Version Number

2024.5.29.27

### What Platforms does this occur on?

Windows

### What headset if any do you use?

desktop

### Log Files

[DESKTOP-LQAQ7BB - 2024.5.29.27 - 2024-05-28 21_58_32.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/15477738/DESKTOP-LQAQ7BB.-.2024.5.29.27.-.2024-05-28.21_58_32.log)

**Comment by gameboycjp** [comment]
They also clip through other objects. 
![2024-05-29 01 45 22](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/37276103/abaaa45d-29b7-4c9a-85e8-0f7557741b8b)
![2024-05-29 01 57 48](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/37276103/2f21adcd-8e8c-4017-b96d-985ff9b141e8)

**Comment by gameboycjp** [comment]
I think it's a regression with "Changed the OffsetFactor/Units for UI materials based on community feedback to prevent ZFighting on UI's when further away from the origin" in 2024.5.29.27. It was 1 and 10 before, right? I think 100 and 1000 is a good bit too much everywhere but the main dashboard because the camera flattens everything there. I tried 10 and 100 on my facet anchors, and it seemed to be a good balance. I'd need to see it with other UI panels to know for sure though.

**Comment by RustybotPrime** [comment]
This also seriously affects protoflux readability as wires appear over nodes even when physically behind them.

**Comment by Frooxius** [comment]
Hmm this is unfortunate. 

I've tweaked the values based on the recommended values by users in #1292, which were apparently tested in practice, but it seems they do actually cause side issues.

The quickest thing we can do is revert them, but that will also reintroduce the Z-fighting issues for now.

**Comment by gameboycjp** [comment]
I highly suggest trying 10 and 100. It'll should still have some reduction in Z-fighting without causing this issue to nearly the same extent.

**Comment by Frooxius** [comment]
Have you tested those values extensively? I'm a bit worried that they will cause issues still.

**Comment by gameboycjp** [comment]
Not extensively. It's not the easiest thing to test as I'm not sure where the background materials are on most UI panels. As much as I would like the old values back, that's gonna cause z-fighting again.

**Comment by TisFoolish** [comment]
Isn't the original issue caused by the the panels scale being insanely small, caused by the Size in the canvas component being treated as pixels instead of meters?  What about adjusting the the Size float2 directly by scaling it down by like a factor of a thousand?

**Comment by Frooxius** [comment]
@gameboycjp I can try the values, but it's possible these problems will still persist.

@TisFoolish That won't work and will cause a myriad of other issues. The size isn't treated like pixels, it's all relative.

**Comment by gameboycjp** [comment]
Yeah, it feels like both issues are symptoms of how UI is rendered. I feel like finding a balance is a decent temporary solution until a proper solution can be worked on later.

**Comment by TisFoolish** [comment]
What i mean is that the person who designed the ui treated it as pixels but the scale is in meters, meaning that setting a canvas' size to 1x1 results in a 1 meter square

**Comment by Frooxius** [comment]
I've made tweaks in 2024.5.31.37, let me know if this works better!

I'm going to close this now, if there are still issues though, we'll reopen.

**Comment by ohzee00** [comment]
I think this might of been under tweaked in the latest patch.

In the world I made here the issue happens again, granted slightly less so (After you change the Offset Factor to 10 and Offset Units to 100 to match the new defaults): https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/2141#issuecomment-2138244387

My suggestion, and with some testing that I did in the previous issue, I think a Offset Factor of 10 and a Offset Units of 1000 can be a really good middle ground of helping z-fighting and https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/2141. Of course, others can try their own values and see too.

So overall my suggestion is keeping the current factor but increasing the Offset Units to 1000.
